### PR TITLE
Send test analytics to dedicated suites for unit, UI iOS, and UI iPadOS

### PR DESCRIPTION
**Pull request by mokagio**
### Description

I created three Test Analytics suites for unit, UI iOS, and UI iPadOS tests to collect more granular info. In particular, given the difference in test run time between unit and UI, having them split means we should have more accurate average figures.

> **Info** There's some issue with UI tests with I haven't yet been able to solve. I'd still like to merge this PR as it is, so that we can have more accurate unit tests metrics in the meantime.

I added a notice on this regard to the build page so we don't forget:

image

### Testing instructions
Verify that analytics have been reported for this build by following the "Test Analytics" link in the [build page](https://buildkite.com/automattic/woocommerce-ios/builds/6498).

---
- [x] I have considered if this change warrants user-facing release notes and have added them to `RELEASE-NOTES.txt` if necessary.

### CODE REVIEW

File: WooCommerce/WooCommerceUITests/UITests.xctestplan [hunk 1]
```diff
@@ -43,6 +43,11 @@
         "value" : "$(BUILDKITE_MESSAGE)"
       }
     ],
+    "targetForVariableExpansion" : {
+      "containerPath" : "container:WooCommerce.xcodeproj",
+      "identifier" : "B56DB3C52049BFAA00D4AA8E",
+      "name" : "WooCommerce"
+    },
```
Comment (mokagio) [inline WooCommerce/WooCommerceUITests/UITests.xctestplan:50]: CI doesn't complain about not having this, but Xcode does.

File: .buildkite/commands/run-ui-tests.sh [hunk 2]
```diff
@@ -24,6 +24,26 @@ install_gems
...
+# First, remove any previous notice to avoid duplication
+context='ctx-ui-tests-notice'
+buildkite-agent annotation remove --context "$context" \
+  || true # `remove` will 401 if there's no annotation, but we don't want to fail because of it
+# Then, print a fresh notice
+buildkite-agent annotate \
+  'Test Analytics for UI tests are currently unavailable' \
+  --style 'info' \
+  --context "$context"
+
```
Comment (AliSoftware) [inline .buildkite/commands/run-ui-tests.sh]: `buildkite-agent annotate` will [already replace/update any existing annotation with the same context](https://buildkite.com/docs/agent/v3/cli-annotate#:~:text=You%20can%20update%20an%20existing%20annotation%27s%20body%20by%20running%20the%20annotate%20command%20again%20and%20provide%20the%20same%20context%20as%20the%20one%20you%20want%20to%20update.%20Or%20if%20you%20leave%20context%20blank%2C%20it%20will%20use%20the%20default%20context.) (except if you use `--append` explicitly). That's the whole point of providing a `--context`, after all, so you can reference/update annotations later.

The only case where removing an existing annotation would be useful is if you don't plan to replace it with a new one. But since here you update the `$context` annotation unconditionally, there's no point on removing it first.

```suggestion
buildkite-agent annotate \
  'Test Analytics for UI tests are currently unavailable' \
  --style 'info' \
  --context "$context"

```

File: WooCommerce/WooCommerceUITests/UITests.xctestplan (hunk 1 above)
Comment (AliSoftware) [reply WooCommerce/WooCommerceUITests/UITests.xctestplan:50]: I'm not sure how this `targetForVariableExpansion` works, but I'm wondering if this might be the reason why UI Tests analytics don't work, especially given that Unit Tests run on their own Xcode target (where they don't depend on the app target and should ideally be totally independent and isolated from the app target?) and use a dedicated test runner, but UI Tests use a Host target (the app) and run inside it like a plugin driving the app via accessibility APIs… so should that `targetForVariableExpansion` here be anything different for UI Tests vs Unit Tests given all that?

_(This is just a total guess and shot in the dark, as I've never played with XCTestPlans and never seen this `targetForVariableExpansion` property in my life before 😅)_

Comment (mokagio) [reply WooCommerce/WooCommerceUITests/UITests.xctestplan:50]: My understanding of the `targetForVariableExpansion` is that it is a similar feature as what we can do with the scheme, where you can let "Test" use the same environment variables as defined in "Run", which helps with DRY.

I haven't had a chance to investigate more on this. For reference, the error I'm getting locally with the UI tests (using an additional env var `BUILDKITE_ANALYTICS_DEBUG_ENABLED`) is

```
[BuildkiteTestCollector] debug: uploading TestResults(format: "json", runEnv: Core.RunEnvironment(ci: Optional("buildkite"), key: "", url: Optional(""), branch: Optional(""), commitSha: Optional(""), number: Optional(""), jobId: Optional(""), message: Optional(""), debug: Optional("true"), version: Optional("0.1.1"), collector: Optional("test-collector-swift")), data: [Core.Trace(id: "68B6B3B6-FA71-4A70-9EE6-D3A70F3924CF", scope: Optional("ReviewsTests"), name: Optional("test_load_reviews_screen"), identifier: Optional("ReviewsTests.test_load_reviews_screen"), location: nil, fileName: nil, result: Optional(Core.Trace.Result.failed), failureReason: Optional("XCTAssertTrue failed - Page UITestsFoundation.PasswordScreen is not loaded."), failureExpanded: [Core.Trace.FailureExpanded(expanded: ["Assertion Failure at BaseScreen.swift:24: XCTAssertTrue failed - Page UITestsFoundation.PasswordScreen is not loaded."], backtrace: ["0 Address 4377259627 is in untracked image \'XCTestCore\'", "1 Address 4377322471 is in untracked image \'XCTestCore\'", "2 Address 4377322819 is in untracked image \'XCTestCore\'", "3 Address 4379884299 is in untracked image \'libXCTestSwiftSupport.dylib\'", "4 Address 4379883619 is in untracked image \'libXCTestSwiftSupport.dylib\'", "5 Address 4414629075 is in untracked image \'UITestsFoundation\'", "6 Address 4414629435 is in untracked image \'UITestsFoundation\'", "7 Address 4414629527 is in untracked image \'UITestsFoundation\'", "8 Address 4379869427 is in untracked image \'libXCTestSwiftSupport.dylib\'", "9 Address 4379869503 is in untracked image \'libXCTestSwiftSupport.dylib\'", "10 Address 7357804347 is in untracked image \'libswiftObjectiveC.dylib\'", "11 Address 4379869139 is in untracked image \'libXCTestSwiftSupport.dylib\'", "12 Address 4414628407 is in untracked image \'UITestsFoundation\'", "13 Address 4414627911 is in untracked image \'UITestsFoundation\'", "14 Address 4414508347 is in untracked image \'UITestsFoundation\'", "15 Address 4414507715 is in untracked image \'UITestsFoundation\'", "16 Address 4414467199 is in untracked image \'UITestsFoundation\'", "17 static LoginFlow.logInWithWPcom()", "18 ReviewsTests.setUpWithError()", "19 @objc ReviewsTests.setUpWithError()", "20 Address 4377472791 is in untracked image \'XCTestCore\'", "21 Address 4377540595 is in untracked image \'XCTestCore\'", "22 Address 4377540311 is in untracked image \'XCTestCore\'", "23 Address 4377274967 is in untracked image \'XCTestCore\'", "24 Address 4377539939 is in untracked image \'XCTestCore\'", "25 Address 4377540467 is in untracked image \'XCTestCore\'", "26 Address 4377472743 is in untracked image \'XCTestCore\'", "27 Address 4377472259 is in untracked image \'XCTestCore\'", "28 Address 4377472015 is in untracked image \'XCTestCore\'", "29 Address 4377242947 is in untracked image \'XCTestCore\'", "30 Address 4377471819 is in untracked image \'XCTestCore\'", "31 Address 4377359307 is in untracked image \'XCTestCore\'", "32 Address 4377505147 is in untracked image \'XCTestCore\'", "33 Address 4377465855 is in untracked image \'XCTestCore\'", "34 Address 4377464939 is in untracked image \'XCTestCore\'", "35 Address 4377471147 is in untracked image \'XCTestCore\'", "36 Address 4377242947 is in untracked image \'XCTestCore\'", "37 Address 4377469691 is in untracked image \'XCTestCore\'", "38 Address 4377360943 is in untracked image \'XCTestCore\'", "39 Address 4377468715 is in untracked image \'XCTestCore\'", "40 Address 4377168739 is in untracked image \'XCTestCore\'", "41 Address 4377373259 is in untracked image \'XCTestCore\'", "42 Address 4377372855 is in untracked image \'XCTestCore\'", "43 Address 4377371327 is in untracked image \'XCTestCore\'", "44 Address 4377360943 is in untracked image \'XCTestCore\'", "45 Address 4377371239 is in untracked image \'XCTestCore\'", "46 Address 4377371991 is in untracked image \'XCTestCore\'", "47 Address 4377168739 is in untracked image \'XCTestCore\'", "48 Address 4377373259 is in untracked image \'XCTestCore\'", "49 Address 4377372855 is in untracked image \'XCTestCore\'", "50 Address 4377371327 is in untracked image \'XCTestCore\'", "51 Address 4377360943 is in untracked image \'XCTestCore\'", "52 Address 4377371239 is in untracked image \'XCTestCore\'", "53 Address 4377371991 is in untracked image \'XCTestCore\'", "54 Address 4377168739 is in untracked image \'XCTestCore\'", "55 Address 4377373259 is in untracked image \'XCTestCore\'", "56 Address 4377372855 is in untracked image \'XCTestCore\'", "57 Address 4377371327 is in untracked image \'XCTestCore\'", "58 Address 4377360943 is in untracked image \'XCTestCore\'", "59 Address 4377371239 is in untracked image \'XCTestCore\'", "60 Address 4377371991 is in untracked image \'XCTestCore\'", "61 Address 4377168739 is in untracked image \'XCTestCore\'", "62 Address 4377175735 is in untracked image \'XCTestCore\'", "63 Address 4377360943 is in untracked image \'XCTestCore\'", "64 Address 4377175463 is in untracked image \'XCTestCore\'", "65 Address 4377597387 is in untracked image \'XCTestCore\'", "66 Address 4377597699 is in untracked image \'XCTestCore\'", "67 Address 4377349955 is in untracked image \'XCTestCore\'", "68 Address 4377212015 is in untracked image \'XCTestCore\'", "69 Address 4377349171 is in untracked image \'XCTestCore\'", "70 Address 4377170243 is in untracked image \'XCTestCore\'", "71 Address 4373075951 is in untracked image \'WooCommerceUITests-Runner\'", "72 Address 4373075923 is in untracked image \'WooCommerceUITests-Runner\'", "73 Address 6445997971 is in untracked image \'CoreFoundation\'", "74 Address 6445994599 is in untracked image \'CoreFoundation\'", "75 Address 6445972011 is in untracked image \'CoreFoundation\'", "76 Address 6445969943 is in untracked image \'CoreFoundation\'", "77 Address 6646265355 is in untracked image \'GraphicsServices\'", "78 Address 6523751063 is in untracked image \'UIKitCore\'", "79 Address 6523770419 is in untracked image \'UIKitCore\'", "80 Address 4373076483 is in untracked image \'WooCommerceUITests-Runner\'", "81 Address 4376501463 is in untracked image \'dyld_sim\'", "82 Address 4374696075 is in untracked image \'dyld\'", "83 Unable to determine owner (executable image) for address 16872595266455207935"])], history: Core.Trace.Span(section: "top", startAt: Optional(304929.4531096667), endAt: Optional(304962.7564847917), duration: Optional(33.30337512498954), detail: [:], children: []))])

[BuildkiteTestCollector] error: The data couldn’t be read because it is missing.
```

I can't make whether the issue is with the error format which doesn't report a test failure but a long list of "Address ... is in untracked image ..." or is due to some env var missing.

My next step will be to try the approach in the simplified environment of [TestPress](https://github.com/automattic/TestPress-Apple).

---

Update: I replicated the setup in the simplified environment of [TestPress](https://github.com/automattic/TestPress-Apple) and I get the same "The data couldn't be read..." error, even when the tests succeed and the [logs](https://buildkite.com/automattic/testpress-apple/builds/30#0181dbf2-f39b-4a3c-bc4b-7840a8a0f141/785-891) show the env vars are forwarded properly.

File: .buildkite/commands/run-ui-tests.sh (hunk 2 above)
Comment (mokagio) [reply .buildkite/commands/run-ui-tests.sh]: Thank you for the info.

I implemented the suggestion in [`589776d`](https://github.com/woocommerce/woocommerce-ios/pull/7163/commits/589776dd429a04ec0be37e4505e8a381b2f5a0ed)

File: .buildkite/commands/run-ui-tests.sh [hunk 3]
```diff
@@ -24,6 +24,19 @@ install_gems
...
+echo "--- :test-analytics: Configure Test Analytics"
+# Collect data separately for iPhone and iPad
+if [[ $DEVICE =~ ^iPhone ]]; then
+  export BUILDKITE_ANALYTICS_TOKEN=$BUILDKITE_ANALYTICS_TOKEN_UI_TESTS_IPHONE
+else
+  export BUILDKITE_ANALYTICS_TOKEN=$BUILDKITE_ANALYTICS_TOKEN_UI_TESTS_IPAD
+fi
+
+buildkite-agent annotate \
+  'Test Analytics for UI tests are currently unavailable' \
```
Comment (AliSoftware) [inline .buildkite/commands/run-ui-tests.sh:36]: I just realized that the wording for this might be confusing — especially for devs not familiar with TA nor with your struggles while working on it.

See https://github.com/wordpress-mobile/WordPress-iOS/pull/19010#discussion_r916564202 for wording change suggestions.


TL;DR one suggestion is having a GitHub issue tracking this with more info about what that means and add a link to it, another is just change the wording to this:

```suggestion
  "Buildkite Test Analytics are currently not reported for UITests. We're investigating why and working on fixing it." \
```

Or you can have a mix of both (ie wording above + link to issue), up to you.

Comment (mokagio) [reply .buildkite/commands/run-ui-tests.sh:36]: It turns out that the issue I thought was only due to my UI tests setup is an issue with the Buildkite library and APIs, see https://github.com/buildkite/test-collector-swift/issues/25.

So, I might sit on this for a few days and hopefully I can just remove the notice altogether.

**Comment by mokagio** [comment]
Closing in favor of #7282
